### ISSUE TITLE
[Bug]: Plugin error Cannot read properties of undefined (reading 'additional_kwargs')

### ISSUE THREAD

**Issue by SailFlorve** [danny-avila/LibreChat#751]
### What happened?

LibreChat      | <-----Agent Model: gpt-3.5-turbo | Temp: 0 | Functions: true----->
LibreChat      | files [
LibreChat      |   'Ai_PDF.json',
LibreChat      |   'Diagrams.json',
LibreChat      |   'Dr_Thoths_Tarot.json',
LibreChat      |   'DreamInterpreter.json',
LibreChat      |   'VoxScript.json',
LibreChat      |   'aitoolhunt.json',
LibreChat      |   'askyourpdf.json',
LibreChat      |   'drink_maestro.json',
LibreChat      |   'earthImagesAndVisualizations.json',
LibreChat      |   'image_prompt_enhancer.json',
LibreChat      |   'qrCodes.json',
LibreChat      |   'rephrase.json',
LibreChat      |   'scholarai.json',
LibreChat      |   'uberchord.json',
LibreChat      |   'web_search.json'
LibreChat      | ]
chat-meilisearch | [2023-08-04T15:27:04Z INFO  actix_web::middleware::logger] 172.27.0.4 "POST /indexes/convos/documents HTTP/1.1" 202 135 "-" "undici" 0.001829
LibreChat      | {
LibreChat      |   Ai_PDF: [AsyncFunction (anonymous)],
LibreChat      |   Diagrams: [AsyncFunction (anonymous)],
LibreChat      |   Dr_Thoths_Tarot: [AsyncFunction (anonymous)],
LibreChat      |   DreamInterpreter: [AsyncFunction (anonymous)],
LibreChat      |   VoxScript: [AsyncFunction (anonymous)],
LibreChat      |   aitoolhunt: [AsyncFunction (anonymous)],
LibreChat      |   askyourpdf: [AsyncFunction (anonymous)],
LibreChat      |   drink_maestro: [AsyncFunction (anonymous)],
LibreChat      |   earthImagesAndVisualizations: [AsyncFunction (anonymous)],
LibreChat      |   image_prompt_enhancer: [AsyncFunction (anonymous)],
LibreChat      |   qrCodes: [AsyncFunction (anonymous)],
LibreChat      |   rephrase: [AsyncFunction (anonymous)],
LibreChat      |   scholarai: [AsyncFunction (anonymous)],
LibreChat      |   uberchord: [AsyncFunction (anonymous)],
LibreChat      |   web_search: [AsyncFunction (anonymous)]
LibreChat      | }
LibreChat      | Requested Tools
LibreChat      | [ 'calculator' ]
LibreChat      | Loaded Tools
LibreChat      | [ 'calculator' ]
LibreChat      | TypeError: Cannot read properties of undefined (reading 'additional_kwargs')
LibreChat      |     at new BaseMessage (/app/node_modules/langchain/dist/schema/index.cjs:22:21)
LibreChat      |     at new AIMessage (/app/node_modules/langchain/dist/schema/index.cjs:79:1)
LibreChat      |     at /app/api/app/clients/PluginsClient.js:245:13
LibreChat      |     at Array.map (<anonymous>)
LibreChat      |     at PluginsClient.initialize (/app/api/app/clients/PluginsClient.js:242:8)
LibreChat      |     at async PluginsClient.sendMessage (/app/api/app/clients/PluginsClient.js:416:5)
LibreChat      |     at async ask (/app/api/server/routes/ask/gptPlugins.js:222:20)
LibreChat      |     at async /app/api/server/routes/ask/gptPlugins.js:67:10

### Steps to Reproduce

choose calculator and ask some questions.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Could be a langchain bug as the calculator tool is native to langchain as well as the functions agent, so the interaction should work fine.

**Comment by xyanglu** [comment]
I'm encountering the same issue, but I'm using a custom tool instead of calculator.

**Comment by danny-avila** [comment]
> I'm encountering the same issue, but I'm using a custom tool instead of calculator.

Is your tool a basic tool or structured? In any case, this is kind of an issue with langchain/the LLM because it may interpret the function wrong.

In any case, if you share your logs it can help me make an issues thread on the langchain library

**Comment by SailFlorve** [comment]
I made modifications to the buildMessages method in OpenAIClient.js and forcefully changed the promptPrefix to a fixed content, which may have affected the pluginClient.js. After reverting the modifications, it started working properly again. 

Since I wanted to provide a non-empty promptPrefix when using the OpenAI endpoint, I made the changes in the if (endpoint === 'azureOpenAI' || endpoint === 'openAI') section of getDefaultConversation.js. Now everything seems to be working fine.
